### DISCUSSION TITLE
Are bundler (vite) aliases necessary?

### DISCUSSION THREAD

**Discussion by szmarci** [moonrepo/moon#1469] [Q&A]
I created a simple monorepo with a single vite app and a single package to test moon and when going through the setup I saw this block: [https://moonrepo.dev/docs/guides/javascript/node-handbook#bundler-integration](https://moonrepo.dev/docs/guides/javascript/node-handbook#bundler-integration)

```typescript
import path from 'path';
import { defineConfig } from 'vite';

export default defineConfig({
  // ...
  resolve: {
    alias: {
      '@company/utils': path.join(__dirname, '../packages/utils/src'),
    },
  },
});
```

Is this really necessary?
I do not have this, nor have I typescript reference paths set up and the application still bundles fine.
I have HMR, it is bundled, it is watched, etc...

Do I miss something if I don't do these setups?

**Comment by milesj** [comment]
@szmarci So are you saying that importing code from a package (not within the vite application) properly triggers HMR? I'd be surprised if that works, since this is a problem for all bundlers, not vite. Perhaps vite added some work arounds for this.

**Reply by szmarci** [reply]
Yes, when I update code inside an imported package I get HMR update from the app in the console:

`9:41:28 AM [vite] hmr update /src/App.tsx`

**Reply by milesj** [reply]
Nice. How's that package.json for the package configured? Is it use `exports` or something that maps to source files? I think that also works.

**Reply by szmarci** [reply]
It's pretty basic:

```json
{
  "name": "@pack/utils",
  "module": "index.ts",
  "type": "module"
}
```

Even if I remove `"module": "index.ts"` it is still working.

**Reply by milesj** [reply]
Not sure what to say. I guess vite supports this natively now.

**Reply by szmarci** [reply]
Maybe I misunderstood this. HMR stops working once you start referencing the artifacts in package.json instead of the source files.
In this case adding the alias resolvers needed for HMR to function.
However, if you don't use it like that, just as an 'internal package' you can get away with zero config like above.

**Reply by muuvmuuv** [reply]
Really wonder what setup you have, if I try that in Angular (with Vite enabled) without building the lib it simply cannot find the libs dependencies in development correctly.
